central/reports/scheduler/v2/scheduler_impl.go

```diff
@@ -427,6 +443,23 @@ func (s *scheduler) validateAndPersistSnapshot(ctx context.Context, snapshot *st
 	return snapshot.GetReportId(), nil
 }
 
+func (s *scheduler) doesViewBasedUserHavePendingReport(userID string) (bool, error) {
+	query := search.NewQueryBuilder().
+		AddExactMatches(search.ReportState, storage.ReportStatus_PREPARING.String()).
```

Comment:
If you don't want the user to queue another job when one is pending, you would also need to check for `storage.ReportStatus.WAITING`